test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -179,3 +191,31 @@ def test_mask_along_axis_iid(self, mask_param, mask_value, axis):
 
         assert mask_specgrams.size() == specgrams.size()
         assert (num_masked_columns < mask_param).sum() == num_masked_columns.numel()
+
+
+@skipIfNoExec('sox')
+class ApplyCodecTestBase(TempDirMixin, TorchaudioTestCase):
+    backend = "sox_io"
+
+    def smoke_test(self, format, compression):
+        """
+        The purpose of this test suite is to verify that apply_codec functionalities do not exhibit
+        abnormal behaviors.
+        """
+        path = self.get_temp_path(f'data.{format}')
+        torch.random.manual_seed(42)
+        waveform = torch.rand(2, 44100 * 1)
+        sample_rate = 8000
+        augmented_data = F.apply_codec(waveform, sample_rate, format=format, channels_first=True,
+                                       compression=compression)
+        save_wav(path, augmented_data, sample_rate)
+        info = sox_io_backend.info(path)
+        assert info.sample_rate == sample_rate
```

Comment:
Since the data is saved with the given `sample_rate`. This assertion always passes. 
Since the `apply_codec` works in-memory manner, it is unnecessary to save data on file system. (`TempDirMixin` is not necessary)

Since it is not simple to check if the resulting data are correct,  I suggest that you check the shape (number of channels and conditionally number of frames).

```python
def _smoke_test(self, format, compression, *, check_num_frames):
    torch.random.manual_seed(42)
    sample_rate = 8000
    num_frames = 3.0 * sample_rate
    num_channels = 2
    waveform = torch.rand(num_channels, num_frames)
    augmented = F.apply_codec(
        waveform, sample_rate, format, channels_first=True, compression=compression)
    assert augmented.dtype == waveform.dtype
    assert augmented.shape[0] == num_channels
    if check_num_frames:
        assert augmented.shape[1] == num_frames
```

then for each formats,

```python
def test_wave(self):
    self._smoke_test("wav", compression=None, check_num_frames=True)

@parameterized.expand([96, ...])
def test_mp3(self, compression):
    self._smoke_test("mp3", compression, check_num_frames=False)
```

Note that WAVE is uncompressed format, so providing `compression` argument makes no difference.

Try to add other formats that are tested in `save_test` file, like `"opus"`. They have different valid value range for compression argument.

Reply:
I have added `vorbis` and `flac`. By adding `opus` i got `RuntimeError: Unsupported file type: opus`?

Reply:
I also got an error inside the unit-test for the cases of mp3 files.
Something like this ```RuntimeError: Error loading audio file: failed to open file.```

Reply:
Oh sorry, I have forgotten that `torchaudio` can only read OPUS but cannot write.